### PR TITLE
Fixed chasms generating too close to mining base enterance some times

### LINKED ISSUE #22310: Chasms generate too close of the mining station  they also can't be fixed using buildmode by placing volcanic floors
**Issue Description**:

Chasms near the mining station's entrance, miners are unable to get out easily. Admins can't place volcanic floors if they try to replace the chasms.

**What did you expect to happen**: 

The chasms  prevented the miners from leaving the mining station and the admins are unable  to replace chasms with volcanic floors.

**What happened instead**:
Miners can't mine, admins can't fix the chasms easilly.

**Why is this bad/What are the consequences**:

Miners can't mine, admins can't fix the chasms easilly.

**Steps to reproduce the problem**:

The chasms blocking mining station is probably RNG and will be hard to reproduce.
To reproduce the buildmode issue: you need to find a chasm turf and create a volcanic floor on it. The tile will be black and unable to be interacted with buildmode.

**When did the problem start happening**:

It happened when the PR randomizing lava was merged.

**Extra information**:

The white floors on the first image is supposed to be the chasms.
image
image

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds a no lava helper to the area near the mining base entrance, to ensure miners can exit the mining base

Partially resolves #22310 // NOTE: Does not resolve build mode shenanigans.

## Why It's Good For The Game

Making miners have to not build a bridge out of the base is good.

## Testing

edited map, ran mapmerge, confirmed turfs had correct flag in game.

## Changelog
:cl:
fix: Fixed chasms generating too close to mining base enterance some times.
/:cl:

### CODE REVIEW

File: code/modules/mod/modules/modules_supply.dm [hunk 1]
```diff
@@ -339,6 +339,8 @@
 	var/speed_up = FALSE
 	if(traveled_tiles == max_traveled_tiles)
 		speed_up = TRUE
+	if(HAS_TRAIT(mod, TRAIT_OIL_SLICKED))
+		speed_up = FALSE
```
Comment (GDNgit) [inline code/modules/mod/modules/modules_supply.dm:343]: uh

**Comment by Contrabang** [comment]
image
Wont the bottom right of this make all the lava that is naturally there turn into ground, making it kind of obvious that its a square there? Maybe remove that bit in the bottom right

**Comment by Qwertytoforty** [comment]
> image Wont the bottom right of this make all the lava that is naturally there turn into ground, making it kind of obvious that its a square there? Maybe remove that bit in the bottom right

Nope! It only stops generation of the rivers there

**Comment by S34NW** [comment]
I would suggest a more "circle" or noisy approach on the edges to make it slightly less obvious if lava does spawn around the box and is just cut off in a long unbroken line

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**